pkg/dao/repositories.go

```diff
@@ -5,52 +5,54 @@ import (
 	"gorm.io/gorm"
 )
 
-type PublicRepository struct {
+// Repository internal (non-user facing) representation of a repository
+type Repository struct {
 	UUID     string
 	URL      string
 	Revision string
 }
 
-func GetPublicRepositoryDao(db *gorm.DB) PublicRepositoryDao {
-	return publicRepositoryDaoImpl{
+func GetRepositoryDao(db *gorm.DB) RepositoryDao {
+	return repositoryDaoImpl{
 		db: db,
 	}
 }
 
-type publicRepositoryDaoImpl struct {
+type repositoryDaoImpl struct {
 	db *gorm.DB
 }
 
-func (p publicRepositoryDaoImpl) FetchForUrl(url string) (error, PublicRepository) {
+func (p repositoryDaoImpl) FetchForUrl(url string) (error, Repository) {
 	repo := models.Repository{}
 	result := p.db.Where("URL = ?", url).First(&repo)
 	if result.Error != nil {
-		return result.Error, PublicRepository{}
+		return result.Error, Repository{}
 	}
-	return nil, PublicRepository{
+	return nil, Repository{
 		UUID:     repo.UUID,
 		URL:      repo.URL,
 		Revision: repo.Revision,
 	}
 }
 
-func (p publicRepositoryDaoImpl) List() (error, []PublicRepository) {
+func (p repositoryDaoImpl) List() (error, []Repository) {
 	var repos []models.Repository
-	var publicRepos []PublicRepository
-	result := p.db.Where("public = true").Find(&repos)
+	var publicRepos []Repository
+	result := p.db.Find(&repos)
 	if result.Error != nil {
 		return result.Error, publicRepos
 	}
 	for i := 0; i < len(repos); i++ {
-		publicRepos = append(publicRepos, PublicRepository{
-			UUID: repos[i].UUID,
-			URL:  repos[i].URL,
+		publicRepos = append(publicRepos, Repository{
+			UUID:     repos[i].UUID,
+			URL:      repos[i].URL,
+			Revision: repos[i].Revision,
 		})
 	}
 	return nil, publicRepos
 }
 
-func (p publicRepositoryDaoImpl) UpdateRepository(pubRepo PublicRepository) error {
+func (p repositoryDaoImpl) Update(pubRepo Repository) error {
```

Comment:
similarly here, rename `pubRepo` to some new name, since it's not a "public repo" anymore